packages/fetch/src/index.ts

```diff
@@ -179,14 +179,31 @@ ${
     isFormUrlEncoded,
   });
 
-  const fetchImplementationBody = mutator
-    ? customFetchResponseImplementation
-    : fetchResponseImplementation;
-
-  const fetchImplementation = `export const ${operationName} = async (${args}): ${retrunType} => {
-  ${bodyForm ? `  ${bodyForm}` : ''}
-  ${fetchImplementationBody}}
-`;
+  let fetchImplementation: string;
+  if (mutator?.isHook) {
+    const fetchImplementationBody = `return ${operationName}(${fetchFnOptions});`;
+    fetchImplementation = `export const use${pascal(operationName)}Hook = (): (${args}) => ${retrunType} => {
+      const ${operationName} = ${mutator.name}<Awaited<${retrunType}>>();
```

Comment:
Since `operationName` is confusing, how about naming it something like `customHook`, `customMutator`, or `mutator`, which makes it clear that it is a custom hook?

```suggestion
      const customHook = ${mutator.name}<Awaited<${retrunType}>>();
```